### ISSUE TITLE
Final review of FMI2.0.1 RC 

### ISSUE THREAD

**Issue by chrbertsch** [modelica/fmi-standard#623]
**Please inspect the PDF documents of the RC (see below) and comment in case of errors or possible improvements - ASAP but before Oct 1st.
In case we do not find blockers, we plan to vote on the release in the FMI steering committee on Oct 2nd.**

//Updated files due to misssing pages
[FunctionalMockupInterface_v2.0.1_RC1.pdf](https://github.com/modelica/fmi-standard/files/3641266/FunctionalMockupInterface_v2.0.1_RC1.pdf)
[FunctionalMockupInterface_v2.0.1_RC1_with_changemarks.pdf](https://github.com/modelica/fmi-standard/files/3641267/FunctionalMockupInterface_v2.0.1_RC1_with_changemarks.pdf)

// Update 2019-09-25 10:43CEST
[FunctionalMockupInterface_v2.0.1_RC2.pdf](https://github.com/modelica/fmi-standard/files/3651462/FunctionalMockupInterface_v2.0.1_RC2.pdf)
[FunctionalMockupInterface_v2.0.1_RC2_with_changemarks.pdf](https://github.com/modelica/fmi-standard/files/3651463/FunctionalMockupInterface_v2.0.1_RC2_with_changemarks.pdf)
[FunctionalMockupInterface_v2.0.1-Diff_RC1_RC2.pdf](https://github.com/modelica/fmi-standard/files/3651464/FunctionalMockupInterface_v2.0.1-Diff_RC1_RC2.pdf)

// Update 2019-09-27 21:22CEST : RC3 added
[FunctionalMockupInterface_v2.0.1_RC3.pdf](https://github.com/modelica/fmi-standard/files/3665911/FunctionalMockupInterface_v2.0.1_RC3.pdf)
[FunctionalMockupInterface_v2.0.1_RC3_with_changemarks.pdf](https://github.com/modelica/fmi-standard/files/3665912/FunctionalMockupInterface_v2.0.1_RC3_with_changemarks.pdf)
[FunctionalMockupInterface_v2.0.1-Diff_RC2_RC3.pdf](https://github.com/modelica/fmi-standard/files/3665913/FunctionalMockupInterface_v2.0.1-Diff_RC2_RC3.pdf)

// Update 2019-10-01 18:41CEST : RC4 added
[FunctionalMockupInterface_v2.0.1_RC4.pdf](https://github.com/modelica/fmi-standard/files/3677583/FunctionalMockupInterface_v2.0.1_RC4.pdf)
[FunctionalMockupInterface_v2.0.1_RC4_with_changemarks.pdf](https://github.com/modelica/fmi-standard/files/3677584/FunctionalMockupInterface_v2.0.1_RC4_with_changemarks.pdf)
[FunctionalMockupInterface_v2.0.1-Diff_RC3_RC4.pdf](https://github.com/modelica/fmi-standard/files/3677585/FunctionalMockupInterface_v2.0.1-Diff_RC3_RC4.pdf)

// Update 2019-10-02  12:24 CEST : RC5 added
[FunctionalMockupInterface_v2.0.1_RC5.pdf](https://github.com/modelica/fmi-standard/files/3680626/FunctionalMockupInterface_v2.0.1_RC5.pdf)
[FunctionalMockupInterface_v2.0.1_RC5_with_changemarks.pdf](https://github.com/modelica/fmi-standard/files/3680628/FunctionalMockupInterface_v2.0.1_RC5_with_changemarks.pdf)
[FunctionalMockupInterface_v2.0.1-Diff_RC4_RC5.pdf](https://github.com/modelica/fmi-standard/files/3680629/FunctionalMockupInterface_v2.0.1-Diff_RC4_RC5.pdf)

// Update 2019-10-06 15:20 CEST: RC6 added (figure numbers fixed)
[FunctionalMockupInterface_v2.0.1_RC6.pdf](https://github.com/modelica/fmi-standard/files/3694391/FunctionalMockupInterface_v2.0.1_RC6.pdf)
[FunctionalMockupInterface_v2.0.1_RC6_with_changemarks.pdf](https://github.com/modelica/fmi-standard/files/3694392/FunctionalMockupInterface_v2.0.1_RC6_with_changemarks.pdf)
[FunctionalMockupInterface_v2.0.1-Diff_RC5_RC6.pdf](https://github.com/modelica/fmi-standard/files/3694393/FunctionalMockupInterface_v2.0.1-Diff_RC5_RC6.pdf)

**Please comment on the latest documents**

**Comment by nickbattle** [comment]
These two documents only go to page 72 or 74 (of 129/137)?

**Comment by chrbertsch** [comment]
>  These two documents only go to page 72 or 74 (of 129/137)?
@nickbattle : Thanks for the hint, I have update the PDFs, now hopefully the all pages are there.

**Comment by nickbattle** [comment]
Thanks. Both are complete now.

**Comment by andreas-junghanns** [comment]
[pages refer to the pages in the document WITH change marks]
 - Cover page: XMLfile -> XML files
 - Cover page: remove "On the" and add "The" instead to fix grammer
 - page 9: ZIPfile -> ZIP file
 - page 9 in 1.1: The grammar of "Expressivity" is now broken: leave original
 - page 19 2nd paragraph from bottom: I n -> In (remove space)
 - page 31 in Table cell for ModelVariable: visibleaccessible looks odd: fix?
 - page 31, last sentence: "in section 0" - there is no section 0, is there?
 - page 36, sentence above table: in the () is a full sentence, start capital and add ´.´
 - page 41, table cell describing unit: move '.' to after "equation" from after '.]' (to be consistent with other cell entries)
 - several ']' are at the next line, first character: odd placement, many others are added at the end of the sentence. Is current use ok?
 - page 49, in description of fmi2SetXXX: new [] contains sentence with missing '.'.
 - page 53, new [], '.' and space after ']' missing
 - page 57, in table description of start, new "WARNING": keywords are wrong font (e.g. modelDescription.xml, fmi2SetXXX)
 - same new insert: C-code -> C code
 - page 63, new sentence: "Model Exchange" correct, or "ModelExchange"? "nx" different font?
 - page 70, last new Note: Why now start with the ':'-style? Both this and previous Notes have '.' at wrong position.
 - page 71, new text in binaries: has two '.' - why?
 - underneath: why here have '>' on new line?
 - page 72, new sentence inserted: odd width of the paragraph

**Comment by andreas-junghanns** [comment]
- page 77, item 4.: space missing before '='
 - why have figure numbers changed? which figure was removed?
 - page 87, new insert: spaces missing around '=' for reinit="true"
 - page 90, last sentence in first new insert: "the stat" -> "the function"
 - same insert: remove '.' after ']'
 - page 90, last new insert that page: fix quotes in variability =/= 'constant"
 - same issue on next page again
 - page 112, in section instantiated: quotes should all be " (replace the bent versions)
 - same place, there is a causality= : insert a space before the '='
 - page 114, footnotes for table: insert spaces around '=' and =/= where missing
        (unless of course this causes a page break in the description)
 - all document: we still have functions with (...) in the document: I thought
   we wanted to use (..)?

**Comment by nickbattle** [comment]
The example XML on p100 (of the changemarked version) has commas after the valueReference attribute of the x[1] and x[2] ScalarVariables. This isn't valid XML:
```
<ScalarVariable name="x[1]" valueReference="0", initial="exact"> <Real/>
</ScalarVariable> <!—index="5" -->
<ScalarVariable name="x[2]" valueReference="1", initial="exact"> <Real/>
</ScalarVariable> <!—index="6" -->
```
Similarly, the abbreviated examples on p66/67 imply wrongly that there are commas between attributes:
```
<ScalarVariable name="p" , ...> ... </ScalarVariable>
```
And I think strictly, the XML encoding attribute on p99 should have a hyphen, "UTF-8".

**Comment by nickbattle** [comment]
Also in the example XML on p99, some of the comments have had the hyphens turned into m-dashes, so a copy from the document fails to parse:
```
<!—index="1" -->  should be <!--index="1" -->
```
Also the example has some SV and unit errors, as shown by fmuCheck and our own VDMCheck tool:
```
$ runFmuCheck.sh example.xml
[INFO][FMILIB] XML specifies FMI standard version 2.0
[INFO][FMI2XML] [Line:74] Detected during parsing:
[ERROR][FMI2XML] Start attribute is required for this causality, variability and initial combination
[INFO][FMI2XML] [Line:77] Detected during parsing:
[ERROR][FMI2XML] Start attribute is required for this causality, variability and initial combination
```
```
$ VDMCheck.sh example.xml                                                                                                                                    
2.2.7 Variable "x[1]" initial/variability/start <exact>/<continuous>/nil invalid at line 77
2.2.7 ScalarVariable "x[1]" invalid at line 77
2.2.7 Variable "x[2]" initial/variability/start <exact>/<continuous>/nil invalid at line 80
2.2.7 ScalarVariable "x[2]" invalid at line 80
2.2.1 ScalarVariables invalid
2.2.3 SimpleType "Modelica.SIunits.Torque", Real unit "N.m" not defined in UnitDefinitions at line 29
2.2.7 ScalarVariable "torque.tau", Real unit "N.m" not defined in UnitDefinitions at line 57
Errors found.
```

**Comment by nickbattle** [comment]
The XML other example on p121 only has the "UTF-8" encoding and N.m units problems:
```
2.2.3 SimpleType "Modelica.SIunits.Torque", Real unit "N.m" not defined in UnitDefinitions at line 29
2.2.7 ScalarVariable "torque.tau", Real unit "N.m" not defined in UnitDefinitions at line 50
```

**Comment by nickbattle** [comment]
Small point, but I note that "?" is included as both a Q-char and an escape in the grammar on p69. Otherwise the sets are disjoint. Is that correct?

**Comment by CThuleHansen** [comment]
On p. 45 regarding VendorAnnotations with type `fmi2Annotation` it is stated `Attribute “name” must be unique with respect to all other elements of the VendorAnnotation list.`
But this is not the case on p. 57 for Annotations with type `fmi2Annotation`. 

This is confusing as the types are the same, `fmi2Annotation` but the invariants are different.

**Comment by CThuleHansen** [comment]
Unnecessary restriction related to fmi2Discard: Figure 12 on p. 107 shows that `fmi2Discard` leads to `stepFailed`. In order to continue simulation from this point, it is necessary to `fmi2reset` of `fmi2SetFMUState` according to table on p. 109. However, it is stated on p. 105: `fmi2Discard – if the slave computed successfully only a subinterval of the communication`

Why is the computed subinterval not valid - in other words, why does it require a reset or setstate in order to continue? Perhaps the restriction can be lifted and the computed subinterval made valid?

**Comment by jph-tavella** [comment]
Some small issues in the changemarks of the text after reading the FunctionalMockupInterface_v2.0.1_RC1_with_changemarks.pdf file.

page 2: rewrite relase in release
page 14: rewrite #indclude in #include
page 19: rewrite I n in In
page 31: rewrite visibleaccessible in visible accessible

in addition, section 0 are visible at different place in the text; should be an error?

**Comment by nickbattle** [comment]
On p21, the fmi2SetDebugLogging function indicates what happens for nCategories > 0, but not when it equals zero. The fmusdk has interpreted this to mean enable or disable all categories. Can that be clarified?

Also, it is not clear whether this is intended to allow toggling of categories (multiple calls to enable/disable category sets) or whether it is a one-shot call. The fmusdk interprets it as one shot, resetting all categories before setting those passed.

This section ends with "See section 0".

**Comment by chrbertsch** [comment]
> 
> 
> Small point, but I note that "?" is included as both a Q-char and an escape in the grammar on p69. Otherwise the sets are disjoint. Is that correct?

@nickbattle or someone else: I am no expert on this and cannot decide this. Could you please come up with a proposal?

**Comment by nickbattle** [comment]
@chrbertsch The files attached now pass VDMCheck and fmuCheck:

[example_3.3.2.xml.txt](https://github.com/modelica/fmi-standard/files/3647588/example_3.3.2.xml.txt)
[example_4.3.2.xml.txt](https://github.com/modelica/fmi-standard/files/3647589/example_4.3.2.xml.txt)

**Comment by nickbattle** [comment]
> > Small point, but I note that "?" is included as both a Q-char and an escape in the grammar on p69. Otherwise the sets are disjoint. Is that correct?
> 
> @nickbattle or someone else: I am no expert on this and cannot decide this. Could you please come up with a proposal?

I think this inconsistency is present in the Modelica 3.2 grammar as well, so there is perhaps a case for leaving it as it is. If we want to clear it up, I suggest that "?" is a Q-char rather than an escape, since the character would not normally require escaping in most other contexts.

**Comment by chrbertsch** [comment]
> > > Small point, but I note that "?" is included as both a Q-char and an escape in the grammar on p69. Otherwise the sets are disjoint. Is that correct?
> > 
> > 
> > @nickbattle or someone else: I am no expert on this and cannot decide this. Could you please come up with a proposal?
> 
> I think this inconsistency is present in the Modelica 3.2 grammar as well, so there is perhaps a case for leaving it as it is. If we want to clear it up, I suggest that "?" is a Q-char rather than an escape, since the character would not normally require escaping in most other contexts.

Ah, there is already a ticket for this, #612 --> Please discuss there.
If this is not solved unexpectedly  soon,  this will not be part of FMI 2.0.1

**Comment by chrbertsch** [comment]
> 
> 
> On p. 45 regarding VendorAnnotations with type `fmi2Annotation` it is stated `Attribute “name” must be unique with respect to all other elements of the VendorAnnotation list.`
> But this is not the case on p. 57 for Annotations with type `fmi2Annotation`.
> 
> This is confusing as the types are the same, `fmi2Annotation` but the invariants are different.

**URGENT:** 
@all : I do not know what to do with this. Please comment and come up with a proposal; otherwise this will not be be included in FMI 2.0.1

**Comment by nickbattle** [comment]
> > On p. 45 regarding VendorAnnotations with type `fmi2Annotation` it is stated `Attribute “name” must be unique with respect to all other elements of the VendorAnnotation list.`
> > But this is not the case on p. 57 for Annotations with type `fmi2Annotation`.
> > This is confusing as the types are the same, `fmi2Annotation` but the invariants are different.
> 
> **URGENT:**
> @ALL : I do not know what to do with this. Please comment and come up with a proposal; otherwise this will not be be included in FMI 2.0.1

VDMCheck verifies this, assuming the tool names should be unique in both places. There are no occurrences of a duplicate name in the whole of the FMI Cross Check repository (792 FMUs), so it might be safe to make that the rule?

**Comment by chrbertsch** [comment]
> 
> 
> Unnecessary restriction related to `fmi2Discard`: Figure 12 on p. 107 shows that `fmi2Discard` leads to `stepFailed`. In order to continue simulation from this point, it is necessary to `fmi2reset` of `fmi2SetFMUState` according to table on p. 109. However, it is stated on p. 105: `fmi2Discard – if the slave computed successfully only a subinterval of the communication`
> 
> Why is the computed subinterval not valid - in other words, why does it require a reset or setstate in order to continue? Perhaps the restriction can be lifted and the computed subinterval made valid?

**URGENT:** 
Could someone familiar with this please comment? Shall we open a separate issue to discuss this?
In case of no fast solution/decision , this will not be included in FMI 2.0.1

**Comment by chrbertsch** [comment]
> On p21, the fmi2SetDebugLogging function indicates what happens for nCategories > 0, but not when it equals zero. The fmusdk has interpreted this to mean enable or disable all categories. Can that be clarified?
> 
> Also, it is not clear whether this is intended to allow toggling of categories (multiple calls to enable/disable category sets) or whether it is a one-shot call. The fmusdk interprets it as one shot, resetting all categories before setting those passed.

**URGENT:** 
@ all  : Please comment and come up with a proposal for a wording. Possibly open a new ticket for this. Otherwise this will not be included in FMI 2.0.1

**Comment by nickbattle** [comment]
> **URGENT:**
> @ all : Please comment and come up with a proposal for [setDebugLogging] wording. Possibly open a new ticket for this. Otherwise this will not be included in FMI 2.0.1

@chrbertsch I'm not confident enough to state the intended semantics. But it's hardly critical, so I'll move this out to a new issue.

**Comment by chrbertsch** [comment]
Clarification added wr.t. to fmiversion string in XML remaining to be "2.0" also for FMI 2.0.1 FMUs added to RC2: 

![image](https://user-images.githubusercontent.com/16363372/65547371-a3fdb380-df19-11e9-891b-682975bfd74a.png)

![image](https://user-images.githubusercontent.com/16363372/65547851-94329f00-df1a-11e9-89f3-9816c36f1f53.png)

I also updated the docuemtation string 
in https://github.com/modelica/fmi-design/blob/master/FMI_2.0.1/fmi2ModelDescription.xsd
![image](https://user-images.githubusercontent.com/16363372/65548549-f2ac4d00-df1b-11e9-91ce-aa1fc3912add.png)

Does this make sense? could someone please comment?

**Comment by andreas-junghanns** [comment]
To leave the version string at "2.0" was what we discussed. I remember the argument that 2.0 importers should be able to read 2.0.1 FMUs and recognise 2.0.1 FMUs as 2.0. Correct?

**Comment by chrbertsch** [comment]
> To leave the version string at "2.0" was what we discussed. I remember the argument that 2.0 importers should be able to read 2.0.1 FMUs and recognise 2.0.1 FMUs as 2.0. Correct?

Yes. Could someone please comment on my proposed wording above?
Does it make sense to modify the documentation string on the xsd as proposed? Wouldn't change functionality comapred to FMI 2.0, but then the xsd-files are not bit identical. But if we do not change it, we have something wrong there.

**Comment by chrbertsch** [comment]
**I uploaded RC2, a comparison with FM2.0 with changemarks and a Diff between RC1 and RC2 --> see main ticket text on top of this page**
@nickbattle , @andreas-junghanns , @andreas-junghanns , @jph-tavella @CThuleHansen : Thanks for reviewing an commenting
I have included all findings from comments above (if I understood them and if they have not raised new issues) and commented on them. There are some questions from my side **which are in bold - please comment on my questions**.

I edited also the frontpage to adapt it to 2.0.1

Some comments raised totally new  topics- I have or will move them to separate issues.
**As it is very late in the process - we plan to vote on the release of FMI 2.0.1 in one weeks time - if no one comes up immediately with a contrete proposal for the wording that gets approved by the design group, we will not include these changes in FMI 2.0.1**

**@ all: Please check RC2** and comment from now on RC2 and mention in your ticket, which PDF you are referring to. Thanks!

**Comment by andreas-junghanns** [comment]
Comment(s) page numbers from the RC1-RC2 compare document:
 - page 49: header of table from next page visible here: force page break before the paragraph before?

**Comment by andreas-junghanns** [comment]
RC 1:
- at the end of the table of content, there is an stray "1." below Appendix B
- the release date at the top of the page is July 12, 2019
- contributors list for 2.0.1 is missing
- page 49, there is a lot of space above the last paragraph of the page (and table)
- page 1 and page 2, release date of 2.0.1 is still July 12, 2019 (won´t it be October 2nd?)

**Comment by chrbertsch** [comment]
> RC 1:
> 
>     * at the end of the table of content, there is an stray "1." below Appendix B _chrbertsch: don't see this in RC2, please check_
> 
>     * the release date at the top of the page is July 12, 2019; _chrbertsch: don't see this in RC2, please check_
> 
>     * contributors list for 2.0.1 is missing  _chrbertsch: please use #633 to collect contributors_
> 
>     * page 49, there is a lot of space above the last paragraph of the page (and table)  _chrbertsch: don't see this in RC2, please check; will check pagebreaks and empty lines for RC3_
> 
>     * page 1 and page 2, release date of 2.0.1 is still July 12, 2019 (won´t it be October 2nd?)  chrbertsch: don't see this in RC2, please check_

@ all:  please comment only on the latest RC, currently RC2. Thanks!

**Comment by beutlich** [comment]
page 68:

> [Note: especially section 4.4.17 states that backslashes "\\" are forbidden as path separator only forward slashes "/" are allowed.]

Should there be a comma or "and" before "only"?

**Comment by beutlich** [comment]
Just as a reminder, since not available in the RC2.

1. Would be good if the final release again contains the intended PDF meta data
   - Titel: Functional Mock-up Interface for Model Exchange and Co-Simulation, Version 2.0.1
   - Author: Modelica Association Project "FMI"

2. Also, paper format should be changed from letter (RC2) back to A4 (as v2.0).

3. Last, not least, the PDF version was increased from 1.6 (as v2.0) to 1.7 (RC2). Not sure, if this really is necessary.

**Comment by andreas-junghanns** [comment]
> 
> 
> > RC 1:
> > ```
> > * at the end of the table of content, there is an stray "1." below Appendix B _chrbertsch: don't see this in RC2, please check_
> > 
> > * the release date at the top of the page is July 12, 2019; _chrbertsch: don't see this in RC2, please check_
> > 
> > * contributors list for 2.0.1 is missing  _chrbertsch: please use #633 to collect contributors_
> > 
> > * page 49, there is a lot of space above the last paragraph of the page (and table)  _chrbertsch: don't see this in RC2, please check; will check pagebreaks and empty lines for RC3_
> > 
> > * page 1 and page 2, release date of 2.0.1 is still July 12, 2019 (won´t it be October 2nd?)  chrbertsch: don't see this in RC2, please check_
> > ```
> 
> @modelica/fmi-design: please comment only on the latest RC, currently RC2. Thanks!

Sorry, I checked, none of the issues reported are in RC2.

**Comment by beutlich** [comment]
* page 70: There is `license.{html,txt}` mentioned in  the text. On page 69 it is `license.{txt,html}`. Could be made consistent.

* page 70: Typo: file**r**s (already in v2.0). Better change

> #include directive with “” should be used for header-filers distributed in the FMU instead of using <…> .

to 

> The #include directives with quotes ("") should be used for header-files distributed in the FMU instead of using angle brackets (<>) .

* page 70: Formatting glitch

> The sub-directory `licenses` can be used to bundle all license files

Looks like this was copied from adoc. Change ``licenses`` to "licenses" in fixed font, similar as "sources" or "documentation" on the same page.

Technically, is there only license.html or license.txt allowed in the licenses folder? What if I have more than one file? Do I need to combine them to a single file or may I several files (say license_lib1.txt and license_lib2.txt) then?

**Comment by beutlich** [comment]
page 5: English grammar issue. Change

> a set of test FMUs are provided

to 

> a set of test FMUs is provided

**Comment by beutlich** [comment]
Another PDF issue: The page links from the table of contents (pages 6 and 7) are no longer clickable in  the RC2 (but were in v2.0).

**Comment by beutlich** [comment]
> // Update 2019-09-25 10:43CEST
> [FunctionalMockupInterface_v2.0.1_RC2.pdf](https://github.com/modelica/fmi-standard/files/3651462/FunctionalMockupInterface_v2.0.1_RC2.pdf)
> [FunctionalMockupInterface_v2.0.1_RC2_with_changemarks.pdf](https://github.com/modelica/fmi-standard/files/3651463/FunctionalMockupInterface_v2.0.1_RC2_with_changemarks.pdf)

I can see changes in FunctionalMockupInterface_v2.0.1_RC2.pdf that are not present/marked in FunctionalMockupInterface_v2.0.1_RC2_with_changemarks.pdf.

**Comment by andreas-junghanns** [comment]
RC2 (comments mostly pages 100 to 126 in reverse because I am reading backwards, sorry):
 - page 57, '.' at end of page should be removed (just like paragraph above)
 - page 126, table entry for "FMU", '.' missing at the end
 - page 109, first paragraph: 'and "C"': we should say: 'and C code'
 - page 108, footnote 8, fmi2Terminated should be fixed-side font
 - page 106, last paragraph: "fmi2DoStep" -> no quotes and fixed-size font
 - page 106, in Mode: paragraph: causality= -> insert space
     same paragraph a little above: fmi2GetXXX should be fixedsize font
     same paragraph: 'Initializtion Mode:" should start at beginning of the line
     paragraph above: initial -> should be fixed-size font
 - page 105: Description cell of fmi2Terminated: true -> should be fixed-size font (should it be fmi2True?)
 - page 104, last sentence is missing the '.'
   same page above: "fmi2OK – if the communication step was computed successfully until its end." -> should end with ',' as the next line continues the sentense
   same page above: fmi2SetFMUState and currentCommunicationPoint should be in fixed-size font
 - page 103: drop the quotes around all the function arguments in their descriptions, e.g. "vr", "order", "nvr", "value", in both function descriptions
 - page 101: lowest left cell in table: insert spaces around '=' in non-footnote usage

General comment: finding key words starting with fmi2 that are not fixed-size font should be "easy" in the source - I think I should stop spamming you here and you try to find those systematically. OK?

**Comment by beutlich** [comment]
page 5: The one link on that page links to `https://fmi-standard.org/downloads):`, which cannot be resolved. In RC2 it correctly links to `https://fmi-standard.org/downloads`.

**Comment by beutlich** [comment]
page 5: The "Library to test connected FMUs" is not available from https://fmi-standard.org/downloads. Not sure, if it shall be added aon the web page or if that tool should be removed from the spec.

**Comment by chrbertsch** [comment]
> I can see changes in FunctionalMockupInterface_v2.0.1_RC2.pdf that are not present/marked in FunctionalMockupInterface_v2.0.1_RC2_with_changemarks.pdf.

We use the comparison mechanism described in https://github.com/modelica/fmi-standard/blob/support/v2.0.x/CONTRIBUTING.adoc to create the document with changemarks.
@beutlich : Are the other changes important? Then please provide an example.

**Comment by chrbertsch** [comment]
from  @andreas-junghanns comment above: 
>   * page 105 of RC2: Description cell of fmi2Terminated: true -> should be fixed-size font (should it be fmi2True?) _chrbertsch: done in RC3_

Now  reads: 
Returns **fmi2True**, if the slave wants to terminate the simulation. Can be called after fmi2DoStep(..) returned fmi2Discard. Use fmi2LastSuccessfulTime to determine the time instant at which the slave terminated. 

Is this correct?

**Comment by chrbertsch** [comment]
@andreas-junghanns : from your comment above on RC2
>     * page 104, last sentence is missing the '.' _chrbertsch: done in RC3_
>       same page above: "fmi2OK – if the communication step was computed successfully until its end." -> should end with ',' as the next line continues the sentense 

Do not understand. This is part of a list of option for "The function returns:" where every sentence ends up with a '.'

**Comment by andreas-junghanns** [comment]
> 
> 
> @andreas-junghanns : from your comment above on RC2
> 
> > same page (104) above: "fmi2OK – if the communication step was computed successfully until its end." -> should end with ',' as the next line continues the sentense
> 
> Do not understand. This is part of a list of option for "The function returns:" where every sentence ends up with a '.'

Mmh, I thought this entire list is one sentence. But I might be wrong.

**Comment by beutlich** [comment]
> @beutlich : Are the other changes important? Then please provide an example.

I do not know. I just observed it and wondered.

**Comment by chrbertsch** [comment]
**I uploaded RC3,** a **comparison with FM2.0** release with changemarks and **a Diff between RC2 and RC3** --> see main ticket text on top of this page

I have included all findings from comments above that have been clarified and commented on them. There are very few small questions open now.

Some comments raised totally new topics- I have or will move them to separate issues.
As it is very late in the process - we plan to vote on the release of FMI 2.0.1 on Wednesday - if no one comes up immediately with a concrete proposal for the wording that gets approved by the design group, we will not include these changes in FMI 2.0.1
Open are: #630, #629

**It would help, if more people go through the tickets list #616 of fixed tickets and check if they are really included correctly in RC3 and if the wording is fine in #616 for the changelog.**

**@ all: Please check RC3 and comment from now on RC3 and mention in your ticket, which PDF you are referring to. Thanks!**

**Comment by beutlich** [comment]
> page 5: The "Library to test connected FMUs" is not available from https://fmi-standard.org/downloads. Not sure, if it shall be added on the web page or if that tool should be removed from the spec.
> _chrbertsch: yes it is available on the Website under downloads, see Test FMUs._

Copied to https://github.com/modelica/FMIModelicaTest. You might want to update the web site.

**Comment by nickbattle** [comment]
The RC PDFs don't have a table of contents that a PDF viewer can understand, whereas the original 2.0 PDF did. This is useful for quickly skipping to a particular section in the document. Is that an option when generating the PDF?

(I'm using Okular on Linux. I can't try this with other PDF viewers to verify)

**Comment by clagms** [comment]
>The RC PDFs don't have a table of contents that a PDF viewer can understand, whereas the original 2.0 PDF did. This is useful for quickly skipping to a particular section in the document. Is that an option when generating the PDF?

I can confirm this on Windows: no bookmarks available.

**Comment by chrbertsch** [comment]
In the RC2.pdf the table of centents was there. @clagms  andn @nickbattle : Could you please confirm?
in RC3 we switched from Adobe PDF for generating the PDF to Word 2016 -->Export --> credate PDF/XPS Docuemnent. There I do not find options to create the bookmarks
![image](https://user-images.githubusercontent.com/16363372/65878102-8c954f00-e38d-11e9-91df-2751e4ec9b1f.png)
Does anyone have hints how one coould do it? Otherwise  I suggest to switch back to Adope PDF printer to generate the PDF

**Comment by chrbertsch** [comment]
Ah, just saw we have to use the "Create bookmarks using .... headings" with this option it is there. 
Will be included in RC4.

**Comment by nickbattle** [comment]
@chrbertsch I don't see contents with either RC1, 2 or 3.

**Comment by chrbertsch** [comment]
> 
> 
> @chrbertsch I don't see contents with either RC1, 2 or 3.

@nickbattle What about this: 
[test_bookmarks.pdf](https://github.com/modelica/fmi-standard/files/3671146/test_bookmarks.pdf)

**Comment by clagms** [comment]
> > @chrbertsch I don't see contents with either RC1, 2 or 3.
> 
> @nickbattle What about this:
> [test_bookmarks.pdf](https://github.com/modelica/fmi-standard/files/3671146/test_bookmarks.pdf)

It's a bit strange on SumatraPDF:
![image](https://user-images.githubusercontent.com/871103/65880012-5659ce80-e391-11e9-957b-d2f2a11b49de.png)
Same thing in Acrobat Reader.

**Comment by nickbattle** [comment]
Yes, I'm seeing the same thing as @clagms: a burst of section "1." empty entries, followed by a plausible contents for the remainder. And "Figure 13" for some reason. This is Okular on Linux.

**Comment by chrbertsch** [comment]
> It's a bit strange on SumatraPDF:

no, get it also with Adobe acrobat 
![image](https://user-images.githubusercontent.com/16363372/65880260-cb2d0880-e391-11e9-985e-98c024345f3a.png)

I get the same with Adobe PDF export. So this must be part information contained in the word document.

Could someone with better Word 2016 knowledge than I have please give a hint?

**Comment by nickbattle** [comment]
The "1s" are all on page 7. This is where the table of contents is located in Word, so I suspect that is causing it. The rogue "Figure 13" and the individuals listed at the end are also being picked up as a "heading". My guess is that these entries (and perhaps the contents too?) have a "Heading" style? It might be possible to change this to something that isn't a heading style?

**Comment by clagms** [comment]
Thank you for these hints @nickbattle !
When exporting the pdf on my machine (word for office 365) I get something that looks a little bit better:
![image](https://user-images.githubusercontent.com/871103/65882455-5bb91800-e395-11e9-936b-c3198e543174.png)

Now trying to figure out which styles are being picked up by word to export the bookmarks.

**Comment by nickbattle** [comment]
@clagms It's probably anything that inherits from "heading" - just a guess though. What style are those names in the appendix?

**Comment by hyumo** [comment]
- Inconsistent XML word format in Example 4 Page 65
- Figure number errors starting from Figure 332, then 443, 554, 665, 776, 887, ...
- In Figure 3 `t` and `E` are overlapping
- Inconsistent syntax highlighting in pseudo code example 3.2.4

**Comment by clagms** [comment]
> @clagms It's probably anything that inherits from "heading" - just a guess though. What style are those names in the appendix?
@nickbattle that's exactly what it is :)

Below are the elements I had to change the style to "normal" in order to get rid of these spurious headers (note the page numbers).
Maybe you can right-click in the Heading 1 style and select all instances:
![image](https://user-images.githubusercontent.com/871103/65882828-1a753800-e396-11e9-9932-d5fa4341ddc9.png)


![image](https://user-images.githubusercontent.com/871103/65882756-ee59b700-e395-11e9-9315-57c6f14e6421.png)
![image](https://user-images.githubusercontent.com/871103/65883716-dedb6d80-e397-11e9-9ac4-12d01c0bf094.png)

Here's the doc and generated pdf, after these issues have been corrected.
[FunctionalMockupInterface.docx](https://github.com/modelica/fmi-standard/files/3671459/FunctionalMockupInterface.docx)
[FunctionalMockupInterface.pdf](https://github.com/modelica/fmi-standard/files/3671460/FunctionalMockupInterface.pdf)

I guess the spurious "1."'s are caused by the same problem.

**Comment by nickbattle** [comment]
@clagms Confirmed that your modified PDF has a correct table of contents, as viewed in Okular. So that's probably the fix you need, @chrbertsch.

**Comment by filip-stenstrom** [comment]
Page reference: RC3

-  **or else** is returned by the fmi2GetXXX functions // what does "else" refer to? p34
- where 𝑢 continuous-time input // --> where 𝑢 is a continuous-time input, p65

**Comment by clagms** [comment]
In the mathematical model of a co-sim FMU:
![image](https://user-images.githubusercontent.com/871103/65893949-2fa79200-e3a9-11e9-9518-cf5504ca3b32.png)
please clarify that $m_{ido}$ stands for the model input derivative order.

**Comment by clagms** [comment]
### Is Feedthrough Supported?
Please clarify the FMI 2.x Co-simulation support for feed-through.

First we need a definition of feedthrough.

Given an Co-simulation FMU instance $c$ , the input $u_c$ feeds-through to output $y_c$ when there exist two different values $v_1, v_2$ and FMU state $s_c$, such that
$$get_c(set_c(s_c, u_c, v_1), y_c) \neq get_c(set_c(s_c, u_c, v_2), y_c)$$
where:

- the FMU state $s_c$ represents the whole FMU state (just like the one used in the FMI get/set state functions)
- $set_c(s_c, u_c, v_1)$ returns an FMU state, that is used in the call to $get_c$, 
- $set_c(s_c, u_c, v_2)$ returns an FMU state, that is used in the call to $get_c$.

Here are some properties of the feed-through:
- When $u_c$  feeds through to $y_c$, we say that there is an algebraic dependency (or algebraic relationship) between $u_c$ and $y_c$. Also, if all other variables are fixed, it means that there is a function $g(u_c)$ such that $y_c = g(u_c)$.
- Usually, when $dy_c/du_c \neq 0$ then we can also conclude that there is feedthrough from $u_c$ to $y_c$.

Note that this definition of feed-through is well accepted, and it means the exact same thing as "algebraic dependency".

Also note that any ODE that has an output equation of the form:
$$\dot{x} = F(x, u)$$
$$y = G(x, u)$$

has feed-through. 

This definition is therefore consistent with [Kubler2000], that RC3 claims to be consistent with.

Now let's look at the RC3 text regarding feedthrough.

Right at the beginning, it says:

```
Support many and nested FMUs: A simulator may run many FMUs in a single simulation run and/or multiple instances of one FMU. The inputs and outputs of these FMUs can be connected with direct feedthrough. Moreover, an FMU may contain nested FMUs.
```

Setting aside the fact that "direct feedthrough" is redundant with "feedthrough", this induces the reader into thinking that FMI CS supports feedthrough. I would argue that supporting feedthrough is a necessary feature to export Modelica models, in

```
Expressivity: The FMI provides the necessary features that Modelica®, Simulink® and SIMPACK® models1 can be transformed to an FMU.
```

Then, in the definition of the causality attribute, the words algebraic relationship are used:

```
"output": The variable value can be used by another model or slave. The algebraic relationship to the inputs is defined via the dependencies attribute of <fmiModelDescription><ModelStructure><Outputs><Unknown>.
```

Leading the reader to believe that any dependency declared in the ModelStructure, is a feedthrough dependency, and therefore satisfies the definition I put above.

In fact, in the definition of the ModelStructure, it says:

```
Attribute dependencies defines the dependencies of the outputs from the knowns at the current super dense time instant in Event and in Continuous-Time Mode (ModelExchange) and at the current Communication Point (CoSimulation).
```

Again, the reader might think that this means there is an algebraic (an instant) dependency between knowns and unknowns.

Thankfully, in the mathematical definition of the FMU for co-simulation, we observe:

![1569853059558](https://user-images.githubusercontent.com/871103/65895581-f9b7dd00-e3ab-11e9-9c4e-121419528fe9.png)

Which tells the reader that there is no feedthrough in that model (i.e., the output $y_{i+1}$ does not depend algebraically on the input $u_{i+1}$).

However, the text immediately after says

```
Definition (4.1) is consistent with the definition of co-simulation by (Kübler, Schiehlen 2000)
```

which is not true. Equation (4.1) clearly forbids feedthrough, and [Kubler2000] has the equation:

![1569853204517](https://user-images.githubusercontent.com/871103/65895602-03414500-e3ac-11e9-9d8b-b6f406f47bb8.png)

(apologies for the scribbles).

This equation clearly shows the feed-through.

Moreover, later in the standard, there is a remark:

```
There is the additional restriction in \emph{slaveInitialized} state that it is not allowed to call fmi2GetXXX functions after fmi2SetXXX functions without an fmi2DoStep call in between
```

Which makes it clear that feedthrough is not supported, which is also claimed in <https://github.com/modelica/fmi-standard/issues/117#issuecomment-430378201>.

If indeed the intent is not to support feedthrough in the FMI co-simulation, then I suggest that the parts I've highlighted above be rewritten to avoid giving this idea.

Moreover, note that the mathematical definition of FMU co-simulation forbids the use of a Gauss-Seidel master algorithm. This is because a Gauss Seidel algorithm will produce an input $u_{i+1}$ to one of the FMUs that is at the timestamp $i+1 = t+H$, violating the assumptions in Equation (4.1). This also means that solving algebraic loops on the output variables is something that requires rollback (whereas, with feedthrough, rollback would not be needed).

To summarize the changes:

- The claim that `The FMI provides the necessary features that Modelica®`  should maybe be restricted to FMI Model Exchange, as I would argue that having feedthrough is an essential feature of Modelica. Even the simplest coupled mechanical models (such as a mass-spring-damper) need feedthrough to be more accurately co-simulated.
- Avoid using the expressions "algebraic relationship", "algebraic dependency", etc. when referring to the actual FMU inputs/outputs. We need to make the distinction between the inputs/outputs of the model inside the FMU, and the inputs/outputs of the FMU. The model may have feedthrough, but the FMU does not have it, so one might as well forget that the model has feedthrough as well.
- Clarify the use of the Model Structure dependencies. Without feed-through, are these only used to know which directional derivatives to ask for? Btw, having a non-zero directional derivative also hints at an algebraic dependency, but maybe we need to make it clear that this is the model dependency, and not the FMU dependency.

I hope this helps! :)



Refs:

[Kubler2000] - Kübler, R., and W. Schiehlen. 2000. “Two Methods of Simulator Coupling.” *Mathematical and Computer Modelling of Dynamical Systems* 6 (2): 93–113. <https://doi.org/10.1076/1387-3954(200006)6:2;1-M;FT093>.

Gomes, Cláudio, Casper Thule, Levi Lucio, Hans Vangheluwe, and Peter Gorm Larsen. 2019. “Generation of Co-Simulation Algorithms Subject to Simulator Contracts.” In *3rd Workshop on Formal Co-Simulation of Cyber-Physical Systems*, to be published. Oslo, Norway.

**Comment by clagms** [comment]
Please clarify the FMI 2.x Co-simulation on what FMUs should assume about the timestamp of their inputs.

Interpolations and Extrapolations are two different function approximation schemes.

The flag canInterpolate inputs may confuse the reader into thinking that an FMU mathematical model requires inputs $u_{i+1}$ before the execution of the doStep function.

This is not the case:
![1569858328676](https://user-images.githubusercontent.com/871103/65896482-9dee5380-e3ad-11e9-898e-d950286a7497.png)

What I think this flag means is that the FMU can approximate the input function over the interval $t$ to $t+H$. In that case, maybe a name "canApproximateInputs" would be more adequate.

Also, a strict interpretation of the mathematical model means that the Gauss Seidel master algorithm is not a valid FMI Cosim algorithm, because such algorithm will compute $u_{i+1}$ at time $t$, before starting the execution of $ɸ_i$.

Also, which inputs are being approximated by the FMU? All continuous inputs? Or just some?

This is because, as stated in Section 4.1.2., an FMU can be a combination of a hybrid ODE and a sampled data system. In that case, it is not necessarily the case that all continuous inputs are being approximated by the FMU.

I would argue that whether an input can be approximated or not is information on each variable (I haven't thought through the exact conditions on that variable), and not information about the FMU.

Thank you very much for all the work on reviewing these issues! :)

**Comment by beutlich** [comment]
* page 124: could use the new repository of FMITest, i.e. change the link to https://github.com/modelica/FMIModelicaTest
* page 126: change "w.r.t to" to "w.r.t."

**Comment by clagms** [comment]
In the description of "unbounded", it says 
> during time integration, the variable gets a value much larger than its nominal value nominal.
but the meaning of "much larger" is not specified anywhere else. 
How does one know that an FMU implements this attribute correctly?

**Comment by nickbattle** [comment]
> **or else** is returned by the fmi2GetXXX functions // what does "else" refer to? p34

The wording is a bit awkward, but I read it to mean that either a value is set (with fmi2SetXXX) or "else" retrieved (with fmi2GetXXX) relative to the unit specified. I think it would mean the same if the "else" was removed.

**Comment by chrbertsch** [comment]
> 
> 
> > **or else** is returned by the fmi2GetXXX functions // what does "else" refer to? p34
> 
> The wording is a bit awkward, but I read it to mean that either a value is set (with fmi2SetXXX) or "else" retrieved (with fmi2GetXXX) relative to the unit specified. I think it would mean the same if the "else" was removed.

I undstand this sentence in a way that both setting and getting the varibal is w.r.t to the unit. So I would replace the "or else" by "and"

If a variable is associated with a Unit, then the value of the variable has to be provided with the fmi2SetXXX functions **and** is returned by the fmi2GetXXX functions with respect to this Unit.

Please comment, if you agree or disagree with this understanding and proposal.

**Comment by chrbertsch** [comment]
> In the description of "unbounded", it says
> 
> > during time integration, the variable gets a value much larger than its nominal value nominal.
> 
> But the meaning of "much larger" is not specified anywhere else.
> How does one know that an FMU implements this attribute correctly?
@clagms : Even though this is not clearly defined, we cannot fix this in a bugfix release 2.0.1 by clearly defining it,  as this could invalidate existing 2.0 implementations. 
If you consider this topic important, please open a new ticket for FMI 3.0.

**Comment by filip-stenstrom** [comment]
> > > **or else** is returned by the fmi2GetXXX functions // what does "else" refer to? p34
> > 
> > 
> > The wording is a bit awkward, but I read it to mean that either a value is set (with fmi2SetXXX) or "else" retrieved (with fmi2GetXXX) relative to the unit specified. I think it would mean the same if the "else" was removed.
> 
> I undstand this sentence in a way that both setting and getting the varibal is w.r.t to the unit. So I would replace the "or else" by "and"
> 
> If a variable is associated with a Unit, then the value of the variable has to be provided with the fmi2SetXXX functions **and** is returned by the fmi2GetXXX functions with respect to this Unit.
> 
> Please comment, if you agree or disagree with this understanding and proposal.

This makes sense to me

**Comment by antvl** [comment]
Regarding clarifications of the mathematical model for Co-simulation:

- $m_{ido}$ stands for the the model input derivative order.  This should be added to page 105, at the end of the following paragraph:

>   The current input variables ...   (in case of continuous-time variables),
>   where $m_{ido}$ stands for the the model input derivative order.
> 

  As already stated in this paragraph extrapolation only applies to
  continuous-time input variables. Is is clear enough?

- The mathematical model does not prevent implementing a Gauss-Seidel   co-simulation schema. Within this schema, the approximation of  input variables is considered as interpolation.  With Jacobi schema, the approximation is done using extrapolation.  As FMI only describes the interface between a master environment and  a slave component, the mathematical model does not describe the whole  co-simulation schema applied to a set of coonected FMUs, but it does  not prevent applying either Jacobi or Gauss-Seidel algorithms.  Unfortunately, it is not possible in FMI 2.0.1 version, that is
  a maintenance version, to change the wording from "canInterpolateInputs"  to "canApproximateInputs".

- Regarding direct feedthrough, one must distinguish between the FMU for Co-simulation and the model inside the FMU (that is described  by an ODE).  The inside model can have direct feedthroughs between input and output,  and this can be queried using the directional derivatives  (since this applied to the underlying model, not the FMU component).  In contrast, co-simulation, is a numerical approximation that is  applied to the model, and as such, it has been decided that in FMI 2.0,  direct feedthrough is not allowed **at co-simulation level**.  This is expressed by the current mathematical model, and it is actually  a restriction with respect to the schema described by Kübler & Schiehlen.

**Comment by clagms** [comment]
Thank you for the thorough reply, and for paying attention to this topic @antvl !

>As already stated in this paragraph extrapolation only applies to continuous-time input variables. Is is clear enough?

Unfortunately, it is still not enough, as one can have continuous time variable that are being approximated, while others are not. Or does the standard constraint the implementation of the FMU to either implement approximation in all continuous time variables, or on none? If so, please maybe make this clearer to tool providers.

> Within this schema, the approximation of input variables is considered as interpolation. 

Maybe I did not explain this properly. Consider an FMU A, that implements an input **interpolation** scheme. When FMU A is at time $t$, and a setReal is called in that FMU A, the FMU A will assume that it is receiving an input that was computed by another FMU B that is already at time $t+H$.
This assumption has to be made because there is no way of knowing the timestamp of an input when a call is made to setReal.
We may argue that the timestamp of a value corresponds to the timestamp of the internal state of the FMU that computed it (although one has to be careful with this definition, and more details are given in the paper cited below).
If that is correct (I cannot think of any other way of defining the timestamp of a value, but I'm all ears for alternatives!), then the Co-simulation model should depict a "+1" on the $u$ var, as shown in blue:
![image](https://user-images.githubusercontent.com/871103/65947247-27e2fe80-e438-11e9-9dca-dfc8823c02ff.png)
This shows that the gauss seidel is not allowed by the math description of the standard.

Now I try to show why, even if it is allowed, there is an ambiguity in the standard.

When a doStep is called, FMU A will compute an interpolation (think of a linear interpolation, for instance).

Now imagine the exact same FMUs A and B being used in a Jacobi co-simulation. That same FMU A is still making the assumption that the value it gets is from a "future" timestamp, and will consequently perform an interpolation on it.
However, when a Jacobi cosim is used, any approximation on the inputs needs to be an extrapolation.
Without explicit timestamps on the inputs, the implementation of a interpolation is different than the implementation of an extrapolation.
The following figure is an attempt at depicting the difference on the inputs that FMU A will give to its internal numerical solver, when used correct (top) and incorrectly (bottom):
![image](https://user-images.githubusercontent.com/871103/65947617-f454a400-e438-11e9-8d56-bf07829a829b.png)

I also attach the paper where me and the co-authors have hopefully explain this matter a little bit better 😋

**Comment by antvl** [comment]
- The FMI for Co-simulation standard does not require that providing derivatives for all continuous-time input variables is mandatory. It is up to the master environment to provide these derivatives through appropriate calls to the _fmi2SetRealInputDerivatives_ function. The FMU has to deal with that, input variables can have different extrapolation or interpolation orders (as shown on page 108).

We could add the following sentence:

> Differents input variables may have different interpolation order.

- As I said before, the mathematical model does not try to describe all possible co-simulation 
 schemas. It is about the internal representation of co-simulation in the FMU, not about how connected FMUs are evaluated in a particular co-simulation algorithm.
Formulas for extrapolation or interpolation are not part of the 2.0 specification document, as it was the case in the FMI for Co-simulation 1.0. This is to let freedom to the FMU implementer. However, extrapolation or interpolation can be correctly implemented by providing the appropriate values for successive derivatives, as shown on page 47, appendix D, of the FMI for Co-simulation 1.0.1 document.

**Comment by CThuleHansen** [comment]
See https://github.com/modelica/fmi-standard/issues/440#issuecomment-536975569

**Comment by chrbertsch** [comment]
>     * Inconsistent XML word format in Example 4 Page 65

Corrected to 
![image](https://user-images.githubusercontent.com/16363372/65955488-8a43fb00-e448-11e9-9236-583be7f80809.png)
@hyumo please check. If you have findings please  propose the correct formatting

>     * Figure number errors starting from Figure 332, then 443, 554, 665, 776, 887, ...

Figure 332 --> changed to Figure 3, 443 --> 4, 554 --> 5, 665 --> 6  and all other occurences that I found 
> 
>     * In Figure 3 `t` and `E` are overlapping
Will be fixed in RC4


>     * Inconsistent syntax highlighting in pseudo code example 3.2.4

fixed in RC4, only "// comments"  see. e.g.
![image](https://user-images.githubusercontent.com/16363372/65961037-085ace80-e456-11e9-829d-c99f9ab1f070.png)

**Comment by chrbertsch** [comment]
> > If a variable is associated with a Unit, then the value of the variable has to be provided with the fmi2SetXXX functions **and** is returned by the fmi2GetXXX functions with respect to this Unit.
... 
> This makes sense to me

Will change to "and" in RC4

**Comment by GallLeo** [comment]
Would be nice to put the new "Modelica Association" and "FMI" logos on the title page?
https://github.com/modelica/BackOffice/tree/master/Task006_CI/Designer/Logo/Print
(Project leaders do have access to this repo; I'll store the new logos in the general MA repo, as soon as we have it set up)

**Comment by chrbertsch** [comment]
@GallLeo 
![image](https://user-images.githubusercontent.com/16363372/66029926-8a9fcd00-e500-11e9-9e8c-09c745501a4b.png)
included in RC5

**Comment by chrbertsch** [comment]
Changes RC4--> RC5:

Add new MA and FMI Logo.
Add missing #392 to word document

Known issue in RC5: figure numbers

**Comment by beutlich** [comment]
@chrbertsch Can you please edit the issue topic since Oct. 1st is in past.

**Comment by chrbertsch** [comment]
RC5 was accepted on Oct 2nd by the FMI steering committee (with the addition that formatting changes may be done before the release process within MA will be initiated).

The currently last known formatting issues is the figure numbering. 
The root cause has been identified by an incorrect change in numbering already detected in comment https://github.com/modelica/fmi-standard/issues/623#issuecomment-534034916 (@andreas-junghanns , you were right and I corrected my comment there) in combination with a bug of word regarding figure numbering in combination with tracked changes (https://superuser.com/questions/717022/ms-word-figure-numbers-are-wrong)  This will be fixed in RC6

**Comment by chrbertsch** [comment]
Figure numbers fixed in RC6, see above
